packages/augur-ui/src/modules/modal/add-funds.tsx

```diff
@@ -58,42 +51,41 @@ const addFundsTorus = async (amount, address) => {
 
 export const AddFunds = ({
   autoSelect = false,
-  tokenToAdd = DAI,
 }: AddFundsProps) => {
+  const BUY_MIN = 20;
+  const BUY_MAX = 250;
   const { loginAccount, modal, actions: {closeModal} } = useAppStatusStore();
-  const isRelayDown = false;
-  const { useSigner, initialSwapToken, initialAddFundsFlow } = modal;
-  const showTransfer = modal.showTransfer || false;
+  const { env: config, ethToDaiRate, repToDaiRate } = useAppStatusStore();
 
-    //analyticsEvent: () => dP.track(ADD_FUNDS, {}),
   const closeAction = () => {
     if (modal.cb) {
       modal.cb();
     }
     closeModal();
   };
-    
 
-  const [amountToBuy, setAmountToBuy] = useState(createBigNumber(0));
-  const [isAmountValid, setIsAmountValid] = useState(false);
-  const [poolSelected, setPoolSelected] = useState(false);
-  const { env: config, ethToDaiRate, repToDaiRate } = useAppStatusStore();
   const ETH_RATE = createBigNumber(1).dividedBy(
     ethToDaiRate?.value || createBigNumber(1)
   );
   const REP_RATE = createBigNumber(1).dividedBy(
     repToDaiRate?.value || createBigNumber(1)
   );
-  
+
+  const { useSigner = false,  initialAddFundsFlow = null, initialSwapToken= null, tokenToAdd = DAI } = modal;
   const address = loginAccount.address;
   const accountMeta = loginAccount.meta;
-  const BUY_MIN = 20;
-  const BUY_MAX = 250;
+
+  if (!loginAccount) {
+        return null;
+  }
 
   const usingOnRampSupportedWallet = accountMeta &&
     accountMeta.accountType === ACCOUNT_TYPES.TORUS ||
     accountMeta.accountType === ACCOUNT_TYPES.FORTMATIC;
 
+  const [amountToBuy, setAmountToBuy] = useState(createBigNumber(0));
```

Comment:
useStates/hooks should always be as high up as possible. the `if (!loginAccount)` above will cause issues if this exits early and this hook is unreachable. Always put hooks as high as they can go and never have a way for the component to return before it would run each hook.